### ISSUE TITLE
[Bug]: Prompt travel doesn't work properly

### ISSUE THREAD

**Issue by speshiou** [continue-revolution/sd-webui-animatediff#308]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### Have you read FAQ on README?

- [X] I have updated WebUI and this extension to the latest version

### What happened?

The output looks mixup all prompts that are specified for different frames.

Prompt:
```
girl,
0: in jungle
8: in car
16: in the space
24: close up
```
Output:
The prompt "close up" should gradually affect the output near the 24th frame, but it turns out that it affects the entire video. Besides, the scenery prompt (jungle, car, space) got mixed up.
![00004-893662313](https://github.com/continue-revolution/sd-webui-animatediff/assets/6083887/f8d81843-c204-4512-80b7-927e98af39ce)

Extension settings:
<img width="634" alt="截圖 2023-11-13 上午10 32 48" src="https://github.com/continue-revolution/sd-webui-animatediff/assets/6083887/fb2c45a2-ce68-4d4f-b264-adc920142694">

### Steps to reproduce the problem

1. Launch SD WebUI
2. Enable animatediff extension
3. Enter prompt
4. Generate the video

### What should have happened?

The output videos should be more controllable by input prompt

### Commit where the problem happens

webui: v1.6.0
extension: v1.11.1

### What browsers do you use to access the UI ?

Google Chrome

### Command Line Arguments

```Shell
python launch.py --xformers --share
```

### Console logs

```Shell
2023-11-13 02:19:00,290 - AnimateDiff - INFO - AnimateDiff process start.
2023-11-13 02:19:00,310 - AnimateDiff - INFO - Injecting motion module mm_sd_v15_v2.ckpt into SD1.5 UNet middle block.
2023-11-13 02:19:00,310 - AnimateDiff - INFO - Injecting motion module mm_sd_v15_v2.ckpt into SD1.5 UNet input blocks.
2023-11-13 02:19:00,310 - AnimateDiff - INFO - Injecting motion module mm_sd_v15_v2.ckpt into SD1.5 UNet output blocks.
2023-11-13 02:19:00,311 - AnimateDiff - INFO - Setting DDIM alpha.
2023-11-13 02:19:00,316 - AnimateDiff - INFO - Injection finished.
2023-11-13 02:19:00,316 - AnimateDiff - INFO - Hacking loral to support motion lora
2023-11-13 02:19:00,316 - AnimateDiff - INFO - Hacking CFGDenoiser forward function.
2023-11-13 02:19:00,316 - AnimateDiff - INFO - Hacking ControlNet.
2023-11-13 02:19:00,317 - AnimateDiff - INFO - You are using prompt travel.
  0% 0/20 [00:00<?, ?it/s]
  5% 1/20 [00:06<02:01,  6.38s/it]
 10% 2/20 [00:10<01:31,  5.09s/it]
 15% 3/20 [00:14<01:18,  4.63s/it]
 20% 4/20 [00:18<01:11,  4.46s/it]
 25% 5/20 [00:23<01:05,  4.39s/it]
 30% 6/20 [00:27<01:02,  4.43s/it]
 35% 7/20 [00:32<00:58,  4.48s/it]
 40% 8/20 [00:36<00:52,  4.35s/it]
 45% 9/20 [00:40<00:46,  4.24s/it]
 50% 10/20 [00:44<00:42,  4.22s/it]
 55% 11/20 [00:48<00:38,  4.29s/it]
 60% 12/20 [00:53<00:34,  4.33s/it]
 65% 13/20 [00:57<00:30,  4.33s/it]
 70% 14/20 [01:01<00:25,  4.25s/it]
 75% 15/20 [01:05<00:20,  4.19s/it]
 80% 16/20 [01:09<00:16,  4.16s/it]
 85% 17/20 [01:14<00:12,  4.17s/it]
 90% 18/20 [01:18<00:08,  4.20s/it]
 95% 19/20 [01:22<00:04,  4.14s/it]
100% 20/20 [01:24<00:00,  4.21s/it]
2023-11-13 02:20:28,222 - AnimateDiff - INFO - Restoring ControlNet.
2023-11-13 02:20:28,222 - AnimateDiff - INFO - Restoring CFGDenoiser forward function.
2023-11-13 02:20:28,222 - AnimateDiff - INFO - Restoring hacked lora
2023-11-13 02:20:28,223 - AnimateDiff - INFO - Restoring DDIM alpha.
2023-11-13 02:20:28,223 - AnimateDiff - INFO - Removing motion module from SD1.5 UNet input blocks.
2023-11-13 02:20:28,223 - AnimateDiff - INFO - Removing motion module from SD1.5 UNet output blocks.
2023-11-13 02:20:28,223 - AnimateDiff - INFO - Removing motion module from SD1.5 UNet middle block.
2023-11-13 02:20:28,223 - AnimateDiff - INFO - Removal finished.
2023-11-13 02:20:28,223 - AnimateDiff - INFO - Merging images into GIF.
2023-11-13 02:20:31,407 - AnimateDiff - INFO - AnimateDiff process end.

Total progress: 100% 20/20 [01:26<00:00,  4.35s/it]
```

### Additional information

Since the prompt travel functionality of this extension is referenced to [animatediff-cli-prompt-travel](https://github.com/s9roll7/animatediff-cli-prompt-travel) (#71), so I also tried the cli tool to inference videos based on the same prompt and model. However, the output video is more controllable by input prompt while using animatediff-cli-prompt-travel.

Prompt:
```
"head_prompt": "girl,",
"prompt_map": {
    "0": "swimsuit, in jungle, from side",
    "8": "office suit, in car, from above",
    "16": "angry, from below, in the space",
    "24": "smile, close up"
},
"tail_prompt": "",
```

Output:
The most input prompt properly affects the specified frame.
![00_673204422663900682_girl__swimsuit_in-jungle_from-side](https://github.com/continue-revolution/sd-webui-animatediff/assets/6083887/df672e5d-f823-49c2-8291-aaf2f5d5f4d9)

I think this symptom may be related to the denoising strength issue which was mentioned by https://github.com/continue-revolution/sd-webui-animatediff/issues/96#issuecomment-1807022562

**Comment by continue-revolution** [comment]
Cli has identical context generator as my code. Actually, everyone including me and comfy copied context generator code from cli. There is no way that I can do about it because even if you choose R-P, some contexts will have end->start. You can try “N” (absolutely no closed loop at all) developed by me, and see what’s going on.

Also, another possibility is that, some samplers in webui actually forward twice the number of forward steps compared to diffusers. The reason is that “step” in A1111 means the steps of sampler execution, but “step” in diffusers is the number of unet forward. We consider the meaning of “step” with some samplers (such as DPM++ SDE Karas, that require two unet forwards) in diffusers a completely bad design choice but we have no way to do about it. If you are using such sampler, you should reduce the number of steps to 10 in webui (in your case, if you are using such sampler) to have a fair comparison to cli.

**Comment by speshiou** [comment]
I tried to reduce the number of steps and use a different sampler, but it didn't work out. I also tried using "N" to avoid getting stuck, but unfortunately, it didn't work either.

Anyway, I appreciate your suggestions.

By the way, I noticed that the author of cli has made some improvements on prompt control (https://github.com/s9roll7/animatediff-cli-prompt-travel/commit/7a9564255a50e747b0a598108fc4643b2e39feb1). 
I'm not sure if that would be helpful.
